src/DSP/AnalogFilter.cpp

```diff
@@ -383,24 +388,74 @@ void AnalogFilter::singlefilterout(float *smp, fstage &hist,
     }
 }
 
-void AnalogFilter::filterout(float *smp)
+void AnalogFilter::singlefilterout_freqbuf(float *smp, fstage &hist,
+                                   float *freqbuf)
 {
-    for(int i = 0; i < stages + 1; ++i)
-        singlefilterout(smp, history[i], coeff);
+    assert((buffersize % 8) == 0);
 
-    if(needsinterpolation) {
-        //Merge Filter at old coeff with new coeff
-        float ismp[buffersize];
-        memcpy(ismp, smp, bufferbytes);
+    float frequency = -1.0f;
+    
+    for ( int i = 0; i < buffersize; i += 8 )
+    {
+	/* recompute coeffs for each 8 samples */
+
+	const float f = ceilf(freqbuf[i] * MAX_FREQ);
+
+	if ( fabsf( f - frequency ) >= 1.0f )
+	{
+	    /* don't perform computation more often than necessary */
+	    computefiltercoefs(f,q);
+	    frequency = f;
+	}
+
+	if(order == 1) {  //First order filter
+	    for ( int j = 0; j < 8; j++ )
+	    {
+		float y0 = smp[i+j] * coeff.c[0] + hist.x1 * coeff.c[1]
+		    + hist.y1 * coeff.d[1];
+		hist.y1 = y0;
+		hist.x1 = smp[i+j];
+		smp[i+j]  = y0;
+	    }
+	} else if(order == 2) {//Second order filter
+
+	    const float coeff_[5] = {coeff.c[0], coeff.c[1], coeff.c[2],  coeff.d[1], coeff.d[2]};
+	    float work[4]  = {hist.x1, hist.x2, hist.y1, hist.y2};
+
+	    AnalogBiquadFilterA(coeff_, smp[i + 0], work);
+	    AnalogBiquadFilterB(coeff_, smp[i + 1], work);
+	    AnalogBiquadFilterA(coeff_, smp[i + 2], work);
+	    AnalogBiquadFilterB(coeff_, smp[i + 3], work);
+	    AnalogBiquadFilterA(coeff_, smp[i + 4], work);
+	    AnalogBiquadFilterB(coeff_, smp[i + 5], work);
+	    AnalogBiquadFilterA(coeff_, smp[i + 6], work);
+	    AnalogBiquadFilterB(coeff_, smp[i + 7], work);
+
+	    hist.x1 = work[0];
+	    hist.x2 = work[1];
+	    hist.y1 = work[2];
+	    hist.y2 = work[3];
+	}
+    }
 
-        for(int i = 0; i < stages + 1; ++i)
-            singlefilterout(ismp, oldHistory[i], oldCoeff);
+    recompute = true;
+}
 
-        for(int i = 0; i < buffersize; ++i) {
-            float x = (float)i / buffersize_f;
-            smp[i] = ismp[i] * (1.0f - x) + smp[i] * x;
-        }
-        needsinterpolation = false;
+void AnalogFilter::filterout(float *smp)
+{
+    float freqbuf[buffersize];
+
+    if ( freq_smoothing.apply( freqbuf, buffersize, freq * MAX_FREQ_CO ) )
+    {
+	/* in transition, need to do fine grained interpolation */
+	for(int i = 0; i < stages + 1; ++i)
+	    singlefilterout_freqbuf(smp, history[i], freqbuf);
+    }
+    else
+    {
+	/* stable state, just use one coeff */
```

Comment:
Did you use 2 spaces for indentation (also in many other places in this PR)? It should be 4...

Reply:
I don't trouble myself with such things, and get whatever EMacs gives me in terms of indentation. Going against the grain is rather tedious. Is there an EMacs mode line or indent script or something equivalent that I run the code through to automatically comply with the formatting requirements?

Reply:
We don't supply such a script. You can fix it manually, or "allow edits from maintainers" in the PR to let one of us do it manually.

Reply:
We do have some really old scripts. I don't know if they work any more though (it was back when the initial yoshimi fork happened). style.cfg and style.sh in the root directory which uses 'uncrustify'. I'm guessing "allow edits from maintainers" is the simpler option.

Reply:
It appears to me that "allow edits by maintainers" is already checked by default.

Reply:
And FYI, what you're looking at currently is already my best effort to "do it manually." Clearly that's not adequate.

Reply:
So, what do you guys do personally? Do you count spaces manually?

Reply:
I'm not sure about everyone else, but with vim, it's mostly shiftwidth=4, expandtab, textwidth=80 for indentation.

Reply:
It's been a while since I used vim for anything other than editing a configuration file. I'm assuming "shiftwidth=4" means a tab is 4 spaces and "expandtab" means the saved file will contain no tabs? If there are no tabs, then I guess that explains the confusion, because the EMacs default is to indent with tabs.